internal/metamorph/processor.go

```diff
@@ -384,6 +394,37 @@ func (p *Processor) StartSendStatusUpdate() {
 	}()
 }
 
+func (p *Processor) StartCheckingTransactionsInNetwork() {
+	p.waitGroup.Add(1)
+	ticker := time.NewTicker(100 * time.Millisecond)
+	go func() {
+		defer p.waitGroup.Done()
+		for {
+			select {
+			case <-p.ctx.Done():
+				return
+
+			case <-ticker.C:
+				p.announcedTransactionsLock.Lock()
+				for k := 0; k < len(p.announcedTransactions); k++ {
```

Comment:
if you move initialization of index counter above the loop you could simply the code like:
```
var k int
for k=0;k <len(p.announcedTransactions); k++{
    if p.announcedTransactions[k].second < uint64(time.Now().Unix())-1 {
     // request tx (...)
    }else{
       break
    }
}

p.announcedTransactions = p.announcedTransactions[k:]
```